pulp_ansible/app/tasks/collections.py

```diff
@@ -793,14 +828,31 @@ async def _download_unpaginated_metadata(self):
 
             collection_endpoint = f"{root_endpoint}/collections/all/"
             excludes_endpoint = f"{root_endpoint}/excludes/"
+            namespaces_endpoint = f"{root_endpoint}/namespaces/?paginate=false"
             col_downloader = self.remote.get_downloader(
                 url=collection_endpoint, silence_errors_for_response_status_codes={404}
             )
             exc_downloader = self.remote.get_downloader(
                 url=excludes_endpoint, silence_errors_for_response_status_codes={404}
             )
-            tasks = [loop.create_task(col_downloader.run()), loop.create_task(exc_downloader.run())]
-            col_results, exc_results = await asyncio.gather(*tasks, return_exceptions=True)
+            nam_downloader = self.remote.get_downloader(
+                url=namespaces_endpoint, silence_errors_for_response_status_codes={404}
```

Comment:
Iterating through every single namespace on the API seems wrong. We don't want to sync every namespace on the server, just the ones that are associated with the collections that are being downloaded.

It seems like it would be more efficient by including the namespace metadata hash in the unpaginated collection version serializer, compare to the hash that we have locally and then pull the namespace metadata from the server if it has been updated or doesn't exist. That would result in more overall api requests, since you have to make an extra request for each changed namespace, but namespaces don't get updated very often so that should only be a problem on the first sync.